### PR TITLE
Add option to configure max distance 

### PR DESCRIPTION
**Pull request by Thulium-Drake**
This PR adds the ability to configure the Max Distance parameter for Chrony and NTP. This will allow, among others, to sync Linux clients to a Windows NTP server.

### CODE REVIEW

**Comment by mlichvar** [comment]
This could be also useful to tighten the requirements on the distance. I'd suggest to not mention any Windows NTP, just explain what it is configuring and that the default is specific to the NTP provider.

**Comment by Thulium-Drake** [comment]
Updated it! :-)

File: README.md [hunk 1]
```diff
@@ -55,6 +55,10 @@ timesync_dhcp_ntp_servers: no
 # Zero threshold disables all steps.
 timesync_step_threshold: 1.0
 
+# Max Distance to accept updates from timeservers
```
Comment (mlichvar) [inline README.md:58]: Can you please expand it to "Maximum root distance to accept measurements for NTP servers" ?

File: templates/chrony.conf.j2 [hunk 2]
```diff
@@ -45,3 +45,8 @@ commandkey 1
 generatecommandkey
 
 {% endif %}
+{% if timesync_max_distance != 0 %}
+# Limits which updates are accepted by this client.
```
Comment (mlichvar) [inline templates/chrony.conf.j2:49]: For consistency with other comments, can you please change it to the imperative style, e.g. "Limit maximum root distance"?

Comment (Thulium-Drake) [reply templates/chrony.conf.j2]: done!

File: README.md (hunk 1 above)
Comment (Thulium-Drake) [reply README.md]: Done!

Comment (mlichvar) [reply README.md]: I made a typo here. I meant "measurements from NTP servers".

Comment (Thulium-Drake) [reply README.md]: Missed that too.. Fixed! :-)

**Comment by mlichvar** [comment]
LGTM. Please squash before merging.

**Comment by Thulium-Drake** [comment]
Done!

**Comment by Thulium-Drake** [comment]
The CI seems to fail with:

```
   TASK [linux-system-roles.timesync : Set variable `timesync_services` with filtered uniq service names] ***

    fatal: [centos-6]: FAILED! => {"msg": "You need to install \"jmespath\" prior to running json_query filter"}

    fatal: [centos-7]: FAILED! => {"msg": "You need to install \"jmespath\" prior to running json_query filter"}
```

Doesn't sound like anything I caused, or did I? :-)

**Comment by richm** [comment]
> The CI seems to fail with:
> 
> ```
>    TASK [linux-system-roles.timesync : Set variable `timesync_services` with filtered uniq service names] ***
> 
>     fatal: [centos-6]: FAILED! => {"msg": "You need to install \"jmespath\" prior to running json_query filter"}
> 
>     fatal: [centos-7]: FAILED! => {"msg": "You need to install \"jmespath\" prior to running json_query filter"}
> ```
> 
> Doesn't sound like anything I caused, or did I? :-)

No, not sure how this issue sneaked in - fixed in https://github.com/linux-system-roles/timesync/pull/90

**Comment by Thulium-Drake** [comment]
After merging #90 is there any way to restart the CI job? Or can this be merged before that?

**Comment by richm** [comment]
> After merging #90 is there any way to restart the CI job? Or can this be merged before that?

Not sure how to restart travis CI - worst case, you'll have to rebase on top of #90 once it is merged and `git push -f` your commit

**Comment by richm** [comment]
@Thulium-Drake #90 was merged - please rebase your commit and push (unless you know of another way to re-trigger travis ci)

**Comment by Thulium-Drake** [comment]
hmm, not what I intended... let me try that again :-)

**Comment by richm** [comment]
[citest commit:72e046944157ca6ab033733c959dfbb55af48909]

**Comment by Thulium-Drake** [comment]
There seem to be quite some issues in the CI which I cannot relate to the change I made, is that something I need to look into?

The only thing seems to be that Fedora32 seems to be unable to start chronyd, which, if I read the output correctly, still runs with default parameters (which is when this change is not active)

**Comment by richm** [comment]
> There seem to be quite some issues in the CI which I cannot relate to the change I made, is that something I need to look into?
> 
> The only thing seems to be that Fedora32 seems to be unable to start chronyd, which, if I read the output correctly, still runs with default parameters (which is when this change is not active)

Right.  This is a problem with systemd on Fedora 32 because we are using outdated images and/or do not refresh the version of systemd.  https://github.com/ansible/ansible/issues/71528

So if @mlichvar agrees, I think it is ok to merge this PR

**Comment by Thulium-Drake** [comment]
@mlichvar ping

**Comment by mlichvar** [comment]
I agree with the change, but please fix the commit message. It seems it was lost in a rebase/squash.

**Comment by Thulium-Drake** [comment]
Done!

**Comment by Thulium-Drake** [comment]
Ping @mlichvar

**Pull request merged**